### PR TITLE
Viacoin support 

### PR DESCRIPTION
**Pull request by romanornr**
Viacoin does have Merged scrypt mining. 
Unsure if I get it working but I will try. 

This might not be an issue for btcpay right? 
Since the explorer should trust the node.

### CODE REVIEW

**Comment by NicolasDorier** [comment]
You can test if you did things correctly:

Add Viacoin here https://github.com/MetacoSA/NBitcoin/blob/master/NBitcoin.Tests/NodeBuilderEx.cs

Then you can run the tests with

```
dotnet test NBitcoin.Tests.csproj --filter "Altcoins=Altcoins" -p:ParallelizeTestCollections=false --framework netcoreapp2.0
```

inside NBitcoin.Tests folder.

**Comment by NicolasDorier** [comment]
If PoW does not work it is not a big deal, but at least you will know if you did any other stupid error.

File: NBitcoin.TestFramework/WellknownNodeDownloadData.cs [hunk 1]
```diff
@@ -115,6 +115,35 @@ public class LitecoinNodeDownloadData
 			};
 		}
 
+		public class ViacoinNodeDownloadData
```
Comment (NicolasDorier) [inline NBitcoin.TestFramework/WellknownNodeDownloadData.cs]: you forgot to expose it as a property as well. Scroll down this file to see how.

**Comment by NicolasDorier** [comment]
great! did you try to run the Altcoins tests so it catches low hanging mistakes you may have done?

**Comment by NicolasDorier** [comment]
does not pass. One is that the regtest genesis block is wrong, other is NBitcoin which fails to connect (probably wrong network magic).

image

**Comment by romanornr** [comment]
Alright I will try to fix this. 
Going to check out today how I can run these C# tests on my arch Linux machine.

**Comment by NicolasDorier** [comment]
@romanornr no different from how to run any test in .NET Core. Just install .NET Core on your machine and run the command line I documented in NBitcoin.Tests.

**Pull request merged**